From 53cb0c17f68c4b786003d3a504d12ecc8fde27a8 Mon Sep 17 00:00:00 2001
From: ZhongXiu Hao <nmred.hao@gmail.com>
Date: Thu, 5 Nov 2020 15:51:29 +0800
Subject: [PATCH] fixed thrift sync wait fatal error

---
 thrift/lib/cpp2/async/RequestChannel.h | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/thrift/lib/cpp2/async/RequestChannel.h b/thrift/lib/cpp2/async/RequestChannel.h
index 16de012..086ce28 100644
--- thrift/lib/cpp2/async/RequestChannel.h
+++ thrift/lib/cpp2/async/RequestChannel.h
@@ -201,13 +201,13 @@ class ClientSyncCallback : public RequestClientCallback {
   explicit ClientSyncCallback(ClientReceiveState* rs) : rs_(rs) {}
 
   void waitUntilDone(folly::EventBase* evb) {
-    if (evb) {
-      if (!evb->inRunningEventBaseThread() || !folly::fibers::onFiber()) {
-        while (!doneBaton_.ready()) {
-          evb->drive();
-        }
-      }
-    }
+    // if (evb) {
+    //   if (!evb->inRunningEventBaseThread() || !folly::fibers::onFiber()) {
+    //     while (!doneBaton_.ready()) {
+    //       evb->drive();
+    //     }
+    //   }
+    // }
     doneBaton_.wait();
   }
 
-- 
1.8.3.1

